libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/AndroidSpeechPlayer.java

```diff
@@ -45,7 +57,7 @@ public void onInit(int status) {
    */
   @Override
   public void play(String instruction) {
-    if (!isMuted && !TextUtils.isEmpty(instruction)) {
+    if (canPlay(instruction)) {
```

Comment:
What about using a explaining variable instead of a query? It seems that `canPlay` is only called from here.